### ISSUE TITLE
ethernet-connected printer only prints one copy when multiple requested

### ISSUE THREAD

**Issue by bluedarter2** [OpenPrinting/cups-filters#242]
This issue is about the inability to print multiple copies of a document from an Ubuntu 20.04 machine.  I filed an issue on the Ubuntu forum, and someone suggested filing it here as well.
https://ubuntuforums.org/showthread.php?t=2445001

Brother MFC-L8900CDW printer is plugged in to ethernet and assigned a static IP number since it's purchase 1.5 years ago. Works perfectly with existing Macs and Windows boxes.

New install of Ubuntu 20.04 on newly built machine. I downloaded the drivers from Brother (https://support.brother.com/g/b/down...cdw_all&os=128) and installed them.

Printing seems to work fine as long as I only want one copy of a document. If I change the quantity to any more than one, I still only get one copy.

This occurs using the print dialog boxes in all the applications and browsers I've tried so far.
It occurs when I print using the lp command, as in:
lp -n 2 -d MFC-L8900CDW test.txt

I have tried toggling the "collate" option in the print dialog box.
I have tried the following, changing (all copies found) of the MFC-L8900CDW.ppd file as follows:
*cupsManualCopies: false
to
*cupsManualCopies: true

Neither of these things helped.

lpoptions returns:
"copies=1 device-uri=ipp://Brother%20MFC-L8900CDW%20series._ipp._tcp.local/ finishings=3 job-cancel-after=10800 job-hold-until=no-hold job-priority=50 job-sheets=none,none marker-change-time=1591573113 marker-colors=#000000,#00FFFF,#FF00FF,#FFFF00,none,none,n one marker-high-levels=100,100,100,100 marker-levels=-1,-1,-1,-1,-1,95,88 marker-low-levels=10,10,10,10 marker-names='Black\ Toner\ Cartridge,Cyan\ Toner\ Cartridge,Magenta\ Toner\ Cartridge,Yellow\ Toner\ Cartridge,Waste\ Toner\ Box,Belt\ Unit,Drum\ Unit' marker-types=toner,toner,toner,toner number-up=1 printer-commands=AutoConfigure,Clean,PrintSelfTestPage printer-info='Brother MFC-L8900CDW series (cups)' printer-is-accepting-jobs=true printer-is-shared=true printer-is-temporary=false printer-location printer-make-and-model='Brother MFC-L8900CDW CUPS' printer-state=3 printer-state-change-time=1591573113 printer-state-reasons=none printer-type=8523868 printer-uri-supported=ipp://localhost/printers/MFC-L8900CDW
"

Can anyone help with this?
Thanks,
Evelyn

**Comment by PIYUSHgoyal16** [comment]
Hello @bluedarter2, thanks for sharing the bug. Can you please send the PPD file so that we can reproduce the same issue in our machines and test out the shortcomings.

**Comment by bluedarter2** [comment]
[brother-ppd.zip](https://github.com/OpenPrinting/cups-filters/files/4833106/brother-ppd.zip)

I couldn't directly attach it, so I've zip'd the one found in /usr/share/ppd/Brother.

There are 2 other identical files found in /usr/share/cups/model/Brother and /opt/brother/Printers/mfcl8900cdw/cupswrapper

Evelyn

**Comment by zdohnal** [comment]
@bluedarter2 IIRC it is possible to attach a PPD file of your print queue from /etc/cups/ppd - you just have to copy the file and add .txt suffix to the copy. Then github is able to attach the file.

**Comment by bluedarter2** [comment]
[MFC-L8900CDW.ppd.txt](https://github.com/OpenPrinting/cups-filters/files/4835088/MFC-L8900CDW.ppd.txt)

OK, it's attached.  It's the same as the one I gave you earlier.

**Comment by zdohnal** [comment]
@bluedarter2 And if your printer is just year and half old, it should be capable of a driverless printing solution (AirPrint, IPP Everywhere). You can try if driverless does help you in any way (plus in driverless guys here have more possibilities to fix it rather than in 3rd part driver from Brother).

You can try driverless printing if you install your printer like this:

$ sudo lpadmin -p <choose_the_name> -v  ipp://Brother%20MFC-L8900CDW%20series._ipp._tcp.local/ -m driverless:ipp://Brother%20MFC-L8900CDW%20series._ipp._tcp.local/ -E

**Comment by zdohnal** [comment]
*cupsFilter: "application/vnd.cups-postscript 0 brother_lpdwrapper_mfcl8900cdw"
*cupsFilter: "application/vnd.cups-pdf 0 brother_lpdwrapper_mfcl8900cdw"

@bluedarter2 that's what I feared - brother driver needs their specific filter. Would you mind trying driverless solution? @PIYUSHgoyal16 can be help you more with that than with Brother driver.

**Comment by bluedarter2** [comment]
> @bluedarter2 And if your printer is just year and half old, it should be capable of a driverless printing solution (AirPrint, IPP Everywhere). You can try if driverless does help you in any way (plus in driverless guys here have more possibilities to fix it rather than in 3rd part driver from Brother).
I use the printer with AirPrint, from a Windows 10 box, and CUPS on my Mac Mini *without problems* (an older version of CUPS on MacOS 10.13.6).

This problem ONLY happens on my new Ubuntu machine.

> $ sudo lpadmin -p <choose_the_name> -v ipp://Brother%20MFC-L8900CDW%20series._ipp._tcp.local/ -m driverless:ipp://Brother%20MFC-L8900CDW%20series._ipp._tcp.local/ -E

I did this and it changed nothing.  Still only 1 copy of the document when I set the quantity to 2.
Yes I'm positive I printed to the newly added "printer".

**Comment by zdohnal** [comment]
> I use the printer with AirPrint, from a Windows 10 box, and CUPS on my Mac Mini _without problems_ (an older version of CUPS on MacOS 10.13.6).
> 
> This problem ONLY happens on my new Ubuntu machine.

Well, you don't use Brother drivers for Linux there, are you? :) Vendors usually don't update their drivers much, which can be a problem on Linux, which is fast developed.

> 
> > $ sudo lpadmin -p <choose_the_name> -v ipp://Brother%20MFC-L8900CDW%20series._ipp._tcp.local/ -m driverless:ipp://Brother%20MFC-L8900CDW%20series._ipp._tcp.local/ -E
> 
> I did this and it changed nothing. Still only 1 copy of the document when I set the quantity to 2.
> Yes I'm positive I printed to the newly added "printer".

Please check the new ppd and attach it here - it is more likely to reproduce it with a file which doesn't require 3rd party software. And CUPS debug log during printing is helpful too - enable it with ```cupsctl --enable-logging```, try to print again and put logs as a file here (I'm not sure what Ubuntu uses for cups log capture - it will be /var/log/cups/error_log or journal -> accessible via journalctl).

Thank you in advance!

**Comment by HonzaK94** [comment]
https://github.com/OpenPrinting/cups-filters/issues/255

**Comment by PIYUSHgoyal16** [comment]
@bluedarter2 Thank you very much for sharing the PPD file. I am trying to reproduce the bug and will check for any bug. I try to report ASAP.

**Comment by bluedarter2** [comment]
Please see this issue:
> #255

**Comment by tillkamppeter** [comment]
Fixed in commit 6157b8f617. Thank you very much for the bug report. Please test.

**Comment by bluedarter2** [comment]
I tried to test this by doing the following:
downloaded the current cups-filters-master sources
configure/make
replaced /usr/lib/cups/filter/gstoraster with the just-built executable

Then I retried printing multiple copies to the MFC-L8900CDW, still only get 1 copy.

I don't want to over-write my whole Ubuntu 20.04 cups filters installation with what I just built, only what is necessary to get this to work correctly, and I am new to Ubuntu and cups, could someone help me by telling me which files should be replaced with my just-built ones?  Thanks.

**Comment by tillkamppeter** [comment]
Sorry, seems that your problem is not the same as #255. There the input file was PostScript and my fix only solves a problem of certain applications sending PostScript jobs. Re-opening.

**Comment by rauferd** [comment]
I have a similar issue with an EPSON ET 3700 network-attached printer on Debian 10 buster, using CUPS 2.2.10 / cupsfilters 1.21.6 and the attached PPD file.

Printing a single copy works fine. Printing multiple copies reports success, but prints only one copy (both lp -n ... and gui print dialog).

Printing multiple copies works fine with Windows or Debian 9 stretch.

I am positive that it is not related to issue #255, since
a) only one copy is printed, even if I replace gstopdf by an old copy of pstopdf in /usr/lib/cups/filter
b) printing does/did work just fine using gstopdf with Debian 9 / CUPS 2.2.1 / cupsfilters 1.11.6

Should I post more information here (which)? Or should I rather open a new issue?


P.S.: Could not attach the PPD file. Please find below the contents of /etc/cups/ppd/EPSON_ET_3700.ppd :

```
*PPD-Adobe: "4.3"
*FormatVersion: "4.3"
*FileVersion: "2.2"
*LanguageVersion: English
*LanguageEncoding: ISOLatin1
*PSVersion: "(3010.000) 0"
*LanguageLevel: "3"
*FileSystem: False
*PCFileName: "ippeve.ppd"
*Manufacturer: "EPSON"
*ModelName: "ET-3700 Series"
*Product: "(ET-3700 Series)"
*NickName: "ET-3700 Series - IPP Everywhere"
*ShortNickName: "ET-3700 Series - IPP Everywhere"
*ColorDevice: True
*cupsVersion: 2.2
*cupsSNMPSupplies: False
*cupsLanguages: "en"
*cupsFilter2: "image/jpeg image/jpeg 0 -"
*cupsManualCopies: true
*cupsFilter2: "image/urf image/urf 100 -"
*OpenUI *PageSize: PickOne
*OrderDependency: 10 AnySetup *PageSize
*DefaultPageSize: A4
*PageSize Letter: "<</PageSize[612 792]>>setpagedevice"
*PageSize 4x6: "<</PageSize[288 432]>>setpagedevice"
*PageSize 5x7: "<</PageSize[360 504]>>setpagedevice"
*PageSize 8x10: "<</PageSize[576 720]>>setpagedevice"
*PageSize 101.6x180.6mm: "<</PageSize[288 511.937007874016]>>setpagedevice"
*PageSize A4: "<</PageSize[595.275590551181 841.889763779528]>>setpagedevice"
*PageSize A5: "<</PageSize[419.527559055118 595.275590551181]>>setpagedevice"
*PageSize A6: "<</PageSize[297.637795275591 419.527559055118]>>setpagedevice"
*PageSize Legal: "<</PageSize[612 1008]>>setpagedevice"
*PageSize B5: "<</PageSize[515.905511811024 728.503937007874]>>setpagedevice"
*PageSize B6: "<</PageSize[362.834645669291 515.905511811024]>>setpagedevice"
*PageSize 3.5x5: "<</PageSize[252 360]>>setpagedevice"
*PageSize Postcard: "<</PageSize[283.464566929134 419.527559055118]>>setpagedevice"
*PageSize Env10: "<</PageSize[296.985826771654 684]>>setpagedevice"
*PageSize EnvC6: "<</PageSize[323.149606299213 459.212598425197]>>setpagedevice"
*PageSize EnvDL: "<</PageSize[311.811023622047 623.622047244094]>>setpagedevice"
*PageSize 5x8: "<</PageSize[360 576]>>setpagedevice"
*PageSize FanFoldGermanLegal: "<</PageSize[612 936]>>setpagedevice"
*PageSize 195x270mm: "<</PageSize[552.755905511811 765.354330708661]>>setpagedevice"
*CloseUI: *PageSize
*OpenUI *PageRegion: PickOne
*OrderDependency: 10 AnySetup *PageRegion
*DefaultPageRegion: A4
*PageRegion Letter: "<</PageSize[612 792]>>setpagedevice"
*PageRegion 4x6: "<</PageSize[288 432]>>setpagedevice"
*PageRegion 5x7: "<</PageSize[360 504]>>setpagedevice"
*PageRegion 8x10: "<</PageSize[576 720]>>setpagedevice"
*PageRegion 101.6x180.6mm: "<</PageSize[288 511.937007874016]>>setpagedevice"
*PageRegion A4: "<</PageSize[595.275590551181 841.889763779528]>>setpagedevice"
*PageRegion A5: "<</PageSize[419.527559055118 595.275590551181]>>setpagedevice"
*PageRegion A6: "<</PageSize[297.637795275591 419.527559055118]>>setpagedevice"
*PageRegion Legal: "<</PageSize[612 1008]>>setpagedevice"
*PageRegion B5: "<</PageSize[515.905511811024 728.503937007874]>>setpagedevice"
*PageRegion B6: "<</PageSize[362.834645669291 515.905511811024]>>setpagedevice"
*PageRegion 3.5x5: "<</PageSize[252 360]>>setpagedevice"
*PageRegion Postcard: "<</PageSize[283.464566929134 419.527559055118]>>setpagedevice"
*PageRegion Env10: "<</PageSize[296.985826771654 684]>>setpagedevice"
*PageRegion EnvC6: "<</PageSize[323.149606299213 459.212598425197]>>setpagedevice"
*PageRegion EnvDL: "<</PageSize[311.811023622047 623.622047244094]>>setpagedevice"
*PageRegion 5x8: "<</PageSize[360 576]>>setpagedevice"
*PageRegion FanFoldGermanLegal: "<</PageSize[612 936]>>setpagedevice"
*PageRegion 195x270mm: "<</PageSize[552.755905511811 765.354330708661]>>setpagedevice"
*CloseUI: *PageRegion
*DefaultImageableArea: A4
*DefaultPaperDimension: A4
*ImageableArea Letter: "8.503937007874 8.503937007874 603.496062992126 783.496062992126"
*PaperDimension Letter: "612 792"
*ImageableArea 4x6: "8.503937007874 8.503937007874 279.496062992126 423.496062992126"
*PaperDimension 4x6: "288 432"
*ImageableArea 5x7: "8.503937007874 8.503937007874 351.496062992126 495.496062992126"
*PaperDimension 5x7: "360 504"
*ImageableArea 8x10: "8.503937007874 8.503937007874 567.496062992126 711.496062992126"
*PaperDimension 8x10: "576 720"
*ImageableArea 101.6x180.6mm: "8.503937007874 8.503937007874 279.496062992126 503.433070866142"
*PaperDimension 101.6x180.6mm: "288 511.937007874016"
*ImageableArea A4: "8.503937007874 8.503937007874 586.771653543307 833.385826771654"
*PaperDimension A4: "595.275590551181 841.889763779528"
*ImageableArea A5: "8.503937007874 8.503937007874 411.023622047244 586.771653543307"
*PaperDimension A5: "419.527559055118 595.275590551181"
*ImageableArea A6: "8.503937007874 8.503937007874 289.133858267717 411.023622047244"
*PaperDimension A6: "297.637795275591 419.527559055118"
*ImageableArea Legal: "8.503937007874 8.503937007874 603.496062992126 999.496062992126"
*PaperDimension Legal: "612 1008"
*ImageableArea B5: "8.503937007874 8.503937007874 507.40157480315 720"
*PaperDimension B5: "515.905511811024 728.503937007874"
*ImageableArea B6: "8.503937007874 8.503937007874 354.330708661417 507.40157480315"
*PaperDimension B6: "362.834645669291 515.905511811024"
*ImageableArea 3.5x5: "8.503937007874 8.503937007874 243.496062992126 351.496062992126"
*PaperDimension 3.5x5: "252 360"
*ImageableArea Postcard: "8.503937007874 8.503937007874 274.96062992126 411.023622047244"
*PaperDimension Postcard: "283.464566929134 419.527559055118"
*ImageableArea Env10: "8.503937007874 8.503937007874 288.48188976378 675.496062992126"
*PaperDimension Env10: "296.985826771654 684"
*ImageableArea EnvC6: "8.503937007874 8.503937007874 314.645669291339 450.708661417323"
*PaperDimension EnvC6: "323.149606299213 459.212598425197"
*ImageableArea EnvDL: "8.503937007874 8.503937007874 303.307086614173 615.11811023622"
*PaperDimension EnvDL: "311.811023622047 623.622047244094"
*ImageableArea 5x8: "8.503937007874 8.503937007874 351.496062992126 567.496062992126"
*PaperDimension 5x8: "360 576"
*ImageableArea FanFoldGermanLegal: "8.503937007874 8.503937007874 603.496062992126 927.496062992126"
*PaperDimension FanFoldGermanLegal: "612 936"
*ImageableArea 195x270mm: "8.503937007874 8.503937007874 544.251968503937 756.850393700787"
*PaperDimension 195x270mm: "552.755905511811 765.354330708661"
*OpenUI *MediaType: PickOne
*OrderDependency: 10 AnySetup *MediaType
*DefaultMediaType: Stationery
*MediaType Stationery: "<</MediaType(Stationery)>>setpagedevice"
*en_US.MediaType Stationery/Plain Paper: ""
*MediaType PhotographicHighGloss: "<</MediaType(PhotographicHighGloss)>>setpagedevice"
*en_US.MediaType PhotographicHighGloss/High Gloss Photo Paper: ""
*MediaType Photographic: "<</MediaType(Photographic)>>setpagedevice"
*en_US.MediaType Photographic/Photo Paper: ""
*MediaType PhotographicSemiGloss: "<</MediaType(PhotographicSemiGloss)>>setpagedevice"
*en_US.MediaType PhotographicSemiGloss/Semi-Gloss Photo Paper: ""
*MediaType PhotographicGlossy: "<</MediaType(PhotographicGlossy)>>setpagedevice"
*en_US.MediaType PhotographicGlossy/Glossy Photo Paper: ""
*MediaType PhotographicMatte: "<</MediaType(PhotographicMatte)>>setpagedevice"
*en_US.MediaType PhotographicMatte/Matte Photo Paper: ""
*MediaType Envelope: "<</MediaType(Envelope)>>setpagedevice"
*en_US.MediaType Envelope/Envelope: ""
*CloseUI: *MediaType
*OpenUI *ColorModel: PickOne
*OrderDependency: 10 AnySetup *ColorModel
*en_US.Translation ColorModel/Color Mode: ""
*ColorModel RGB: "<</cupsColorSpace 19/cupsBitsPerColor 8/cupsColorOrder 0/cupsCompression 0>>setpagedevice"
*en_US.ColorModel RGB/Color: ""
*ColorModel Gray: "<</cupsColorSpace 18/cupsBitsPerColor 8/cupsColorOrder 0/cupsCompression 0>>setpagedevice"
*en_US.ColorModel Gray/Grayscale: ""
*DefaultColorModel: RGB
*CloseUI: *ColorModel
*OpenUI *Duplex: PickOne
*OrderDependency: 10 AnySetup *Duplex
*en_US.Translation Duplex/2-Sided Printing: ""
*DefaultDuplex: None
*Duplex None: "<</Duplex false>>setpagedevice"
*en_US.Duplex None/Off (1-Sided): ""
*Duplex DuplexNoTumble: "<</Duplex true/Tumble false>>setpagedevice"
*en_US.Duplex DuplexNoTumble/Long-Edge (Portrait): ""
*Duplex DuplexTumble: "<</Duplex true/Tumble true>>setpagedevice"
*en_US.Duplex DuplexTumble/Short-Edge (Landscape): ""
*CloseUI: *Duplex
*cupsBackSide: Rotated
*DefaultResolution: 300dpi
*OpenUI *cupsPrintQuality: PickOne
*OrderDependency: 10 AnySetup *cupsPrintQuality
*en_US.Translation cupsPrintQuality/Print Quality: ""
*DefaultcupsPrintQuality: Normal
*cupsPrintQuality Draft: "<</HWResolution[300 150]>>setpagedevice"
*en_US.cupsPrintQuality Draft/Draft: ""
*cupsPrintQuality Normal: "<</HWResolution[300 300]>>setpagedevice"
*en_US.cupsPrintQuality Normal/Normal: ""
*cupsPrintQuality High: "<</HWResolution[300 300]>>setpagedevice"
*en_US.cupsPrintQuality High/High: ""
*CloseUI: *cupsPrintQuality
```

**Comment by nidhijainpnp** [comment]
Hi @bluedarter2, 
Enable the CUPS debug log during printing with `cupsctl --enable-logging`, try to print again and please attach the logs as a .txt file here.(In Ubuntu,error_log file can be found in /var/logs/cups/error_log)Log files will be helpful for me for better debugging.

**Comment by nidhijainpnp** [comment]
@rauferd , it is possible to attach a PPD file of your print queue - you just have to copy the file and add .txt suffix to the copy.

**Comment by bluedarter2** [comment]
@nidhijainpnp I assume you meant: `cupsctl --debug-logging`
I've attached the resulting error_log.
[cups-error_log.txt](https://github.com/OpenPrinting/cups-filters/files/4903786/cups-error_log.txt)

**Comment by nidhijainpnp** [comment]
@bluedarter2  ,Thanks for sharing the error_log. I am trying to reproduce the bug  and will  report ASAP.
Yes ,it is `cupsctl --debug-logging` ,editted in the comment,

**Comment by nidhijainpnp** [comment]
@bluedarter2 ,Could you please mention your cups-filter version also?
I was working on  your issue from past 2 days and was not able to reproduce the error.But,I have managed to identify the filters(from your log files) used in the below operation to print .txt file are:
>lp -n 2 -d MFC-L8900CDW test.txt

1.  texttopdf 
2. pdftopdf
3. brother_lpdwrapper_mfcl8900cdw (brother driver specific filter)
So,I would suggest you to try to swap your first two filter from newer version of CUPS-Filters .Then tell me,if it works.

**Comment by bluedarter2** [comment]
my-ubuntu:/usr/lib/cups$ dpkg -l | egrep cups-filters
ii  cups-filters                               1.27.4-1                              amd64        OpenPrinting CUPS Filters - Main Package
ii  cups-filters-core-drivers                  1.27.4-1                              amd64        OpenPrinting CUPS Filters - Driverless printing

Please advise precisely how I should go about getting newer versions of CUPS-Filters.

I had downloaded the sources and built them, per https://github.com/OpenPrinting/cups-filters/issues/242#issuecomment-655073971
When I move the original installed /usr/lib/cups/filter/texttopdf and pdftopdf aside, and put my .libs/texttopdf and .libs/pdftopdf in their place, nothing prints - the queue stops.
I don't want to bugger my system, so please give me a process that won't do that.

**Comment by tillkamppeter** [comment]
@bluedarter2, please do the following test without trying to update any parts of your cups-filters installation. Simply start from the situation where you get only 1 copy even if you request more than one.
Stop the CUPS daemon:
```
sudo systemctl stop cups
```
Then edit the file (you need to do this with root rights):
```
/etc/cups/ppd/MFC-L8900CDW.ppd
```
You find a line like
```
*cupsManualCopies: false
```
near the beginning of the file. Change this line to
```
*cupsManualCopies: true
```
Save the file and start CUPS again:
```
sudo systemctl start cups
```
Now try to print again. Are you getting the requested number of copies now?

**Comment by rauferd** [comment]
As posted above, my ppd file `EPSON_ET_3700.ppd` originally had
`*cupsManualCopies: true`
so I just tried changig that to
`*cupsManualCopies: false`
but -- not very surprisingly -- still only a single copy is printed.
( So in case this fixes @bluedarter2 's issue, than my issue is probably not related. )

**Comment by bluedarter2** [comment]
@rauferd tldr https://github.com/OpenPrinting/cups-filters/issues/242#issue-634938353: been there, done that, no fix.

**Comment by bluedarter2** [comment]
@tillkamppeter : is stopping and then restarting the cups daemon necessary?
If not: https://github.com/OpenPrinting/cups-filters/issues/242#issue-634938353 done that edit to the ppd file, no fix.

**Comment by tillkamppeter** [comment]
@rauferd, my comment was only intended for @bluedarter2, I have seen that your PPD already has `*cupsManualCopies: true` and switching this to "false" would not help. You must have another issue. As you have also already checked that it is not issue #255, they only way to find out what your problem is is to get your error_log.

Run the command
```
cupsctl --debug-logging
```
Try to print the multiple-copy job again. Then attach your `/var/log/cups/error_log` file here (remember that you have to rename the file to have a `.txt` extension so that you can attach it here).

**Comment by bluedarter2** [comment]
OK @tillkamppeter  I tried the edit to the ppd file AGAIN, this time stopping cup before edit, then restarting after.
Nothing changed, still 1 copy when 2 were requested.

**Comment by bluedarter2** [comment]
> @rauferd, my comment was only intended for @bluedarter2, I have seen that your PPD already has `*cupsManualCopies: true` and switching this to "false" would not help. You must have another issue. 

@tillkamppeter why is @rauferd issue a different issue? Symptoms are the same so far.

**Comment by tillkamppeter** [comment]
@bluedarter2, @rauferd, you are both hitting a bug in the `pdftopdf` filter (so you have actually both the same problem): the filter is supposed to generate multiple copies (send the document's repeatedly, once per copy, to the next filter) by itself when `*cupsManualCopies: true` is set (then `ppd->manual_copies` is True) but it does not.

A simple test shows it (PDF file edited to contain `*cupsManualCopies: true`):
```
FINAL_CONTENT_TYPE=application/vnd.cups-pdf PPD=MFC-L8900CDW.ppd ./pdftopdf 2 2 2 2 '' FILE.pdf > out.pdf
```
`out.pdf` should contain the pages of `FILE.pdf` 2 times. What I get with the current GIT is the pages only 1 time (1 copy).
@nidhijainpnp, could you have a look into the source code of `pdftopdf`, especially the file `filter/pdftopdf/pdftopdf.cc` to check what is going wrong here? Note that `param.numCopies` is the number of copies to be generated by software (by `pdftopdf` itself) and `param.deviceCopies` is the number of copies supposed to be generated by the printer or driver. `pdftopdf`needs to set these 2 variables correctly before processing the pages.

@bluedarter2, note that there is also a bug in the proprietary Brother driver, as with `*cupsManualCopies: false` in the PPD the driver takes the responsibility to generate the copies, which it does not, and therefore we tries the workaround of `*cupsManualCopies: true` and hit the bug in `pdftopdf`. We will not treat the bug in the Brother driver here as we cannot fix it. We will only fix the bug in `pdftopdf` here.

**Comment by bluedarter2** [comment]
@tillkamppeter thanks for the analysis.  I will keep the `*cupsManualCopies: true` in my ppd files.  As soon as you and @nidhijainpnp have worked out a fix, I will have an effective work-around for the Brother driver issue.

I will also make sure to update my support ticket with Brother with the additional information.

However, once you and @nidhijainpnp have worked out a fix, is there any way you can request that Ubuntu maintainers adopt the fix to `pdftopdf` in an update patch?

Otherwise I would like/need some help to update my installed cups filters `pdftopdf` in a way that doesn't break my system.

**Comment by nidhijainpnp** [comment]
@bluedarter2 ,Please edit `*cupsManualCopies: True` in your ppd file. (Be careful ,it is `True` not `true`.Case-sensitive)
Then try to print again and please tell me ,if you are now able to print more than one copy.

**Comment by nidhijainpnp** [comment]
> @bluedarter2, @rauferd, you are both hitting a bug in the `pdftopdf` filter (so you have actually both the same problem): the filter is supposed to generate multiple copies (send the document's repeatedly, once per copy, to the next filter) by itself when `*cupsManualCopies: true` is set (then `ppd->manual_copies` is True) but it does not.
> 
> A simple test shows it (PDF file edited to contain `*cupsManualCopies: true`):
> 
> ```
> FINAL_CONTENT_TYPE=application/vnd.cups-pdf PPD=MFC-L8900CDW.ppd ./pdftopdf 2 2 2 2 '' FILE.pdf > out.pdf
> ```
> 
> `out.pdf` should contain the pages of `FILE.pdf` 2 times. What I get with the current GIT is the pages only 1 time (1 copy).
> @nidhijainpnp, could you have a look into the source code of `pdftopdf`, especially the file `filter/pdftopdf/pdftopdf.cc` to check what is going wrong here? Note that `param.numCopies` is the number of copies to be generated by software (by `pdftopdf` itself) and `param.deviceCopies` is the number of copies supposed to be generated by the printer or driver. `pdftopdf`needs to set these 2 variables correctly before processing the pages.

@tillkamppeter 
The above two variables were set correctly,But on toggling the `*cupsManualCopies: True` , I was able to print the correct number of copies by the following command:
```
FINAL_CONTENT_TYPE=application/vnd.cups-pdf PPD=/home/nidhi/Desktop/brother-ppd/brother_mfcl8900cdw_printer_en.ppd /usr/lib/cups/filter/pdftopdf 2 2 2 2 '' /usr/share/cups/data/form_english.pdf > out.pdf
```
On further examining the file ppd.c,I found the following if-else block:
```
else if (!strcmp(keyword, "cupsManualCopies"))
      ppd->manual_copies = !strcmp(string, "True"); 
```
Here,we are using strcmp() function to compare the strings,
 I was pondering ,shouldn't we allowed case-insensitive strings also for better user experience?We may use strcmpi() function(similar to strcmp() but allows case insensitive strings also) inplace of strcmp() to compare strings .

**Comment by tillkamppeter** [comment]
The file `ppd.c` (in the `ppd/` subdirectory) is part of libppd, the PPD file handling library. This library is new, I have introduced it some days ago. The code is copied from libcups, where the PPD handling functions are deprecated and will soon get removed, because in the future printer drivers will not use PPDs any more. With libppd I want to conserve these functions for retro-fitting existing (legacy) drivers with PPD files.
The code of libppd is simply copied from libcups, so it contains the same bugs and features and if your distribution's `pdftopdf` filter using libcups shows some odd behavior you can assume the same happens with the current `pdftopdf` using libppd.
@nidhijainpnp, but what you have found in `ppd.c` is not actually a bug. The PPD specification tells that all keywords in PPDs are case-sensitive. So the solution is to take care that when generating or editing PPD files to always use the keywords `True` and `False` for boolean variables, NOT `true` and `false`.

**Comment by tillkamppeter** [comment]
IThanks for your investigations, @nidhijainpnp, I have checked through the whole cups-filters package and the addition of the line `*cupsManualCopies: true` in PPD files, both in libcupsfilters and libppd is the only point where the fact that keywords in PPDs are case-sensitive was overlooked.
I have fixed these points in commit 42868e742b.
As libppd is completely copied from libcups the bug still exists in libcups and would need to get reported there.

**Comment by rauferd** [comment]
Dear all,

I can confirm the output of the abovementioned pdftopdf commands.

Consequently, changing
`cupsManualCopies: true`
to
`cupsManualCopies: True`
in my ppd file EPSON_ET_3700.ppd permanently fixes this issue.

( So after all, it was the same issue as @bluedarter2 's)

I will file a bug against the package providing my ppd file.

Thank you very much for your support and the quick solution!

P.S.: Please consider renaming the issue, so that I can be found more easily by others affected by this issue, e.g.


> "cupsManualCopies: Incorrect capitalization in PPD prevents mutliple copies from being printed (was: ethernet-connected printer only prints one copy when multiple requested)"

**Comment by bluedarter2** [comment]
Hey you guys: strcmp() or strcmpi() are subject to security vulnerabilities.  I think you should use strncasecmp() instead.
I will check that True, rather than true, will work for me tonight, can't do now.
Evelyn

**Comment by bluedarter2** [comment]
@tillkamppeter @nidhijainpnp I confirm that using `True` (rather than `true` or `false`) for `cupsManualCopies` results in the expected number of copies produced.

Thanks much for the work-around!

**Comment by ArnoVersfeld** [comment]
I have had this same problem of printing only one (1) copy of a document instead multiple copies as requested. The printer I use is a Canon MX924.
I found cupsManualCopies in the Canon_MX920_series.ppd file to be "true" instead of "True". I changed it to "True" and the printer now prints multiple copies upon request. Thank you very much ! !

**Comment by jsmeix** [comment]
By the way:
The CUPS documentation
http://www.cups.org/doc/spec-ppd.html#cupsManualCopies
```
cupsManualCopies
...
    *cupsManualCopies: true
```
also needs to be fixed.

**Comment by jsmeix** [comment]
I reported it to current upstream CUPS as
https://github.com/OpenPrinting/cups/issues/93

**Comment by andreblecha** [comment]
My Canon MG7500 series, driverless, cups-filters 1.27.4 in Ubuntu 20.04 had the same problem. 
I edited the file /etc/cups/ppd/Canon_MG7500_series_.ppd and set *cupsManualCopies: True
and THAT'S it. Many thank's

It would be wise to add some sort of conclusion in the HEAD of discussion if an obvious solution has been found for the problem. I personaly dont care to read long discussion to the very end, but most user will ....

**Comment by johnh44** [comment]
Yes this issue was fixed in CUPS ver. 2.3.3op1. However, many OS packages still 
come with older versions of CUPS. Among them are recent Apple and Linux 
operating systems, including the popular Linux Mint 20.1, which come with CUPS 
2.3.1. Until OS versions include newer versions of CUPS (the latest is 2.4.1), 
the problem will persist. 

As many have recommended, editing the .ppd file for your printer to change 
"true" to "True" (oh, the absurdity of source code that would require such a 
change!) does fix the issue -- for a time. But the inability to print more 
than one copy MAY come back if you ever change the printer's properties. 
And it WILL come back if you add a new printer.

Here are the steps to fix the problem for your current printer(s). It must be 
done as the system root user:

sudo su
systemctl stop cups
cd /etc/cups/ppd

While the CUPS service is not running, edit ALL .ppd file in this directory. 
Search for the line "cupsManualCopies: true" and change true to True. Then:

systemctl start cups
exit

**Comment by trdieharder** [comment]
This fix basically worked for me for printing pdf files. I still have to test it for LibreOffice

I actually didn't have the line 

*cupsManualCopies: True

I went into /etc/cups/ppd 

I selected the ppd file according to the printer I added or wanted to use

The I added in the line

*cupsManualCopies: True

I added it just above the line  *cupsLanguages: "en"

This helped. I can not print multiple copies and it can be collated.
